src/OneOffFrequency.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace ipl\Scheduler;
+
+use DateTime;
+use ipl\Scheduler\Contract\Frequency;
+
+class OneOffFrequency implements Frequency
+{
+    public const NO_REPEAT = 'none';
+
+    /** @var DateTime Start time of this frequency */
+    protected $start;
+
+    /** @var DateTime End time of this frequency */
+    protected $end;
+
+    public function __construct(DateTime $at)
+    {
+        $this->startAt($at);
+        $this->endAt(clone $at);
+    }
+
+    public function isDue(DateTime $dateTime = null): bool
```

Comment:
I think it makes sense to remove `$dateTime = null` from the `isDue()` interface method. We never call it without a parameter if I'm not mistaken. Please open a new PR for that.

Reply:
Prior to or after all other PRs have been merged?

Reply:
We can do that afterwards.